### PR TITLE
feat(xplorers-bot-ts): Remove the use of Makefile in favor of Taskfile

### PR DESCRIPTION
**Pull request by boltdynamics**
### Remove the use of Makefile in favor of Taskfile
A taskfile is a text file that defines a set of tasks that can be executed by a task runner. Task files are typically written in YAML, and they can be used to automate a variety of tasks, such as building, testing, and deploying software. The body of the task file contains the commands that are executed when each task is run.

### Benefits of Taskfile
- Taskfile is simpler and easier to learn. Taskfile uses a simpler YAML-based syntax than Makefile's custom syntax. This makes it easier to learn and use Taskfile, even for beginners.
- Taskfile is more flexible. Taskfile can be used to automate a wider range of tasks than Makefile. For example, Taskfile can be used to automate tasks such as running tests, deploying code, and managing dependencies.
- Taskfile is more portable and platform agnostic.
- Taskfile is more actively maintained. Taskfile is actively maintained by a team of developers. This means that new features are being added to Taskfile on a regular basis, and any bugs that are found are being fixed quickly.

### [CI BUILD](https://github.com/xplorer-io/xplorers-bot-ts/actions/runs/5101341553/jobs/9170103407)
image

### Inspired by @pgaijin66 's recent [video](https://www.youtube.com/watch?v=_-bpnCY3-FE&t=667s) and [blog](https://prabeshthapa.medium.com/taskfile-maintainable-and-readable-alternative-to-makefile-1bc5ff6cf842) on Taskfile 🙇

### CODE REVIEW

**Review by pgaijin66 — APPROVED** [review]
Looks good.

File: Taskfile.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      - task: clean
+      - task: install
+      - task: compile
+      - mkdir -p dist
+      - cp -R node_modules src/* package*.json dist
+
+  create-slack-token-secret:
+    desc: Create a secret in GCP Secret Manager to store the slack token. Don't forget to set SLACK_OAUTH_TOKEN env var before running this command.
+    cmds:
+      - echo ${SLACK_OAUTH_TOKEN}
```
[the remark this answers is not in the document]
Comment (boltdynamics) [reply Taskfile.yml]: Nice catch, removed 👍

File: Taskfile.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    desc: Package and zip the application code for deployment
+    cmds:
+      - task: clean
+      - task: install
+      - task: compile
+      - mkdir -p dist
+      - cp -R node_modules src/* package*.json dist
+
+  create-slack-token-secret:
+    desc: Create a secret in GCP Secret Manager to store the slack token. Don't forget to set SLACK_OAUTH_TOKEN env var before running this command.
```
[the remark this answers is not in the document]
Comment (boltdynamics) [reply Taskfile.yml]: Updated

File: Taskfile.yml [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    desc: Clean artifacts and dependencies
+    cmds:
+      - rm -rf dist/
+      - rm -rf {{.FUNCTION_ARTIFACT}}
+      - rm -rf src/index.js src/helpers/*.js index.js
+
+  package:
+    desc: Package and zip the application code for deployment
+    cmds:
+      - task: clean
```
[the remark this answers is not in the document]
Comment (boltdynamics) [reply Taskfile.yml:49]: Hmm, this `clean` is to clean any stale artifacts before executing the rest of the commands, is there a better way to do this?

[the remark this answers is not in the document]
Comment (pgaijin66) [reply Taskfile.yml:49]: ah i see Is there specific reason to clean up here ? I can see you are differing after applying TF.

**Review by pgaijin66 — APPROVED** [review]
Looks good

**Pull request merged**